pkg/linkerd/resource.go

```diff
@@ -0,0 +1,50 @@
+package linkerd
+
+import (
+	"strings"
+
+	"github.com/deislabs/smi-metrics/pkg/mesh"
+
+	"github.com/deislabs/smi-metrics/pkg/prometheus"
+	"github.com/deislabs/smi-sdk-go/pkg/apis/metrics"
+	"github.com/prometheus/common/model"
+)
+
+type resourceLookup struct {
+	Item     *metrics.TrafficMetricsList
+	interval *metrics.Interval
+	queries  map[string]string
+}
+
+func (r *resourceLookup) Get(labels model.Metric) *metrics.TrafficMetrics {
```

Comment:
Do these really belong here? They feel like some generic prometheus tooling, maybe as a separate module?

Reply:
This will be taken up in a separate issue, so that meshes can reuse the prometheus stuff!
#24